command-line/lib/Options/Applicative/Style.hs

```diff
@@ -89,5 +123,35 @@ styleArg = argument (eitherReader reader) $ mempty
         "byron"       -> Right Byron
         "icarus"      -> Right Icarus
         "shelley"     -> Right Shelley
-        "shared"     -> Right Shared
+        "shared"      -> Right Shared
         _             -> Left $ "Unknown style; expecting one of " <> styles'
+
+passphraseReader :: String -> Either String Passphrase
+passphraseReader str =
+    if null str then
+        pure $ FromEncoded mempty
+    else do
+        let wrds = T.words . T.filter noNewline . T.decodeUtf8 . B8.pack $ str
+        case mkSomeMnemonic @'[ 9, 12 ] wrds of
+            Right mw -> pure $ FromMnemonic mw
+            Left _ -> getBase16 <|> getBase58
+  where
+    getBase16 = case fromBase16 (T.encodeUtf8 $ T.pack str) of
+        Right hex -> pure $ FromEncoded hex
+        Left _ -> Left "not hex-encoded bytes"
+    getBase58 = case fromBase58 (T.encodeUtf8 $ T.pack str) of
+        Right bs -> pure $ FromEncoded bs
+        Left _ -> Left "not base58-encoded bytes"
+    noNewline :: Char -> Bool
+    noNewline = (`notElem` ['\n', '\r'])
```

Comment:
I would suggest to not enforce anything particular on the 2nd-factor. Just a plain UTF-8 bytestring. There's no reason to force a particular ascii-friendly encoding nor to use mnemonic for that. I know that's what cardano-wallet's API does but that's a very debatable design decision -- especially in the context of the command-line.

Reply:
I supported plain utf8 text, bytestring hex encoded, bytestring base64 encoded and octet array